### DISCUSSION TITLE
What is  the difference between d and ds[i]

### DISCUSSION THREAD

**Discussion by Tusenka** [prometheus/prometheus#15002] [Q&A]
What is  the difference between d.dir and ds[i].dir and why would we apply it twice?
https://github.com/prometheus/prometheus/blob/main/tsdb/compact.go#L346

**Comment by bboreham** [comment]
Since the `for` loop is over `ds[1:]`, `d` is `ds[i+1]`.

It's hard to spot exactly what is going on; perhaps this code should be rewritten for clarity.
